dask/_task_spec.py

```diff
@@ -199,7 +199,7 @@ def _execute_subgraph(inner_dsk, outkey, inkeys, external_deps):
         final[k] = DataNode(None, v)
     for k, v in external_deps.items():
         final[k] = DataNode(None, v)
-    res = execute_graph(final)
+    res = execute_graph(final, keys=[outkey])
```

Comment:
This is very important. I factored this out in https://github.com/dask/dask/pull/11510 and added a dedicated test for subgraph callables